core/src/main/java/jenkins/websocket/WebSockets.java

```diff
@@ -115,9 +116,9 @@ public void onWebSocketError(Throwable cause) {
                 }
 
                 @Override
-                public void onWebSocketBinary(byte[] payload, int offset, int length) {
+                public void onWebSocketBinary(ByteBuffer data) {
```

Comment:
I am aware of a single CloudBees proprietary plugin that needs to be adapted to this, but otherwise there are no usages of this interface. Seems cleaner to intentionally break the interface to be consistent with Remoting and Jetty rather than to continue to maintain inconsistent interfaces and do unnecessary copies. The single consumer can simply be updated in lockstep with this change.